changes for groovy and catkinized

marc-hanheide opened the pull request:
Tim,

I made some modifications to make it all compile within our catkin workspace in ROS groovy. However, this doesn't work yet correctly, So feel free to disregard this pull request. 
We talked in Stuttgart at ROSCon about our use case: We want to use MongoDB for logging ROS topics in the [STRANDS FP7 project](http://www.strands-project.eu/)
The problem is as follows: I can start `rosrun mongodb_log mongodb_log /odom`, but then no data is actually inserted into the DB and I get trillions of exceptions as below:

```
marc@marcs-pc:~/ros_ws/src/mongodb_log$ rosrun mongodb_log mongodb_log /odom
Adding topic /odom
GENERIC Python logger used for topic /odom
Updated graphs, total queue size 0, dropped 0, took 0:00:00.000831
ACTIVE: WorkerProcess-   0-/odom
/usr/lib/python2.7/dist-packages/pymongo/collection.py:310: RuntimeWarning: couldn't encode - reloading python modules and trying again. if you see this without getting an InvalidDocument exception please see http://api.mongodb.org/python/current/faq.html#does-pymongo-work-with-mod-wsgi
  continue_on_error, self.__uuid_subtype), safe)
InvalidDocument WorkerProcess-   0-/odom@/odom: 

Cannot encode object: genpy.Time[1373487256959625959]
InvalidDocument WorkerProcess-   0-/odom@/odom: 

Cannot encode object: genpy.Time[1373487256972947120]
InvalidDocument WorkerProcess-   0-/odom@/odom: 

Cannot encode object: genpy.Time[1373487256992713451]
InvalidDocument WorkerProcess-   0-/odom@/odom: 
```

Any idea what I'm doing wrong?

Thanks,

Marc

marc-hanheide committed:
made rostime import to conform with groovy system

marc-hanheide committed:
catikinsed

marc-hanheide commented on src/mongodb_log.py:
```diff
@@ -61,7 +61,7 @@
 import rosgraph.masterapi
 import roslib.message
-from roslib.rostime import Time, Duration
+from rospy.rostime import Time, Duration
```
I'm not 100% sure about this modification, but this is how I got it to work... hope it's correct

marc-hanheide commented on package.xml:
```diff
@@ -0,0 +1,67 @@
+  <!-- One maintainer tag required, multiple allowed, one person per tag --> 
+  <!-- Example:  -->
+  <!-- <maintainer email="jane.doe@example.com">Jane Doe</maintainer> -->
+  <maintainer email="marc@hanheide.net">Marc Hanheide</maintainer>
```
I added me as a maintainer, but feel free to add somebody else, please

marc-hanheide commented on CMakeLists.txt:
```diff
@@ -1,32 +1,160 @@
+## Find catkin macros and libraries
+## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
+## is used, also find other catkin packages
+find_package(catkin REQUIRED rospy roslib rosgraph rostopic tf sensor_msgs)
```
I didn't check properly if all these are needed

marc-hanheide commented:
@timn, are you still working on mongodb_log? Or are you simply enjoying a prolonged vacation this summer ;-) ?

timn commented:
Hi Marc.

First, thanks for the pull request. It's great and I want to merge it, but haven't had the time to review it, yet.

And yes, I'm still working on mongodb_log and using it. Lately more on the Fawkes version (http://git.fawkesrobotics.org/fawkes.git/tree/HEAD:/src/plugins/mongodb_log).

Does the updated version remain compatible with fuerte or is there a way to keep it compatible? It's still on some systems we use.

marc-hanheide commented:
Hi,

ah, good! I assume the change isn't compatible with fuerte. I suggest keeping it in a separate branch (a common strategy to have `fuerte_devel` and `groovy_devel`, etc) And, as I said it's not working yet. Do you know if any of the other forks got it to work properly in groovy? I saw https://github.com/code-iai/ros-mongodb_log did include quite some changes but I had no time either to look at them yet.

Cheers

marc-hanheide committed:
squashed commit for changes that make mongodb_log work on groovy

moved conversion to a nicer place
(cherry picked from commit c7c92bbfe879a331a6fc9952f29f35821618ad56)

fixed issue with serialization of time stamps (genpy namespace), deactivated optimized tf logger for now
(cherry picked from commit 30939bb22d0f60621bb07c72ecb448335fa02eea)

Conflicts:

	src/mongodb_log.py

fixed tf recording with optimized C++ logger: convert path from list to string
(cherry picked from commit a7792fea7c7785f1c1eeb9aeeb78245687516a08)

marc-hanheide committed:
moved conversion to a nicer place
(cherry picked from commit c7c92bbfe879a331a6fc9952f29f35821618ad56)

marc-hanheide committed:
fixed issue with serialization of time stamps (genpy namespace), deactivated optimized tf logger for now
(cherry picked from commit 30939bb22d0f60621bb07c72ecb448335fa02eea)

Conflicts:

	src/mongodb_log.py

marc-hanheide committed:
fixed tf recording with optimized C++ logger: convert path from list to string
(cherry picked from commit a7792fea7c7785f1c1eeb9aeeb78245687516a08)

marc-hanheide committed:
Merge branch 'groovy-catkinzed' of github.com:marc-hanheide/mongodb_log into groovy-catkinzed

marc-hanheide commented:
OK, I got it working now on my groovy installation. See if this is of any use for you. But I definitely suggest you do not pull this into master but something like `groovy-devel` if you like.

rmbennett commented:
I've just come across this repository, I'm also using ros-groovy. It would be great to have an "official"  groovy-devel branch.

I've made, I assume at least, similar additions to @marc-hanheide to get this package to work with ros-groovy. Did you add any extra functionality @marc-hanheide when you created the "groovy-catkinized" branch.

marc-hanheide commented:
no, I didn't add anything, really, but in our projects we have
extended/wrapped it quite a bit towards a ros_datacentre:
https://github.com/strands-project/ros_datacentre which still contains
mongodb_log as a catkin package.

On 20 May 2014 15:57, rmbennett notifications@github.com wrote:

> I've just come across this repository, I'm also using ros-groovy. It would
> be great to have an "official" groovy-devel branch.
> 
> I've made, I assume at least, similar additions to @marc-hanheidehttps://github.com/marc-hanheideto get this package to work with ros-groovy. Did you add any extra
> functionality @marc-hanheide https://github.com/marc-hanheide when you
> created the "groovy-catkinized" branch.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/timn/ros-mongodb_log/pull/2#issuecomment-43637091
> .

## 

---

Dr. Marc Hanheide
Senior Lecturer in Computer Science
Room MC3221B
University of Lincoln
Brayford Pool
Lincoln, LN6 7TS
Phone: +44 1522 88 6966
EMail: marc@hanheide.net
Web: http://www.hanheide.net
skype: mah99de

rmbennett commented:
Thanks for the swift reply @marc-hanheide .

What kind of database queries does the ros_datacentre tool allow for? Is there any documentation available as to what it does and how to use it?

Thanks,

marc-hanheide commented:
Nothing beyond the readmes I'm afraid. It's all pretty fresh and some bits
are probably still a bit project specific. But we use it as a general
persistency layer in our systems. Has also support to restore rosparams and
the message store is pretty much a slightly enhanced version of mongodb_log
On 20 May 2014 16:19, "rmbennett" notifications@github.com wrote:

> Thanks for the swift reply @marc-hanheidehttps://github.com/marc-hanheide.
> 
> What kind of database queries does the ros_datacentre tool allow for? Is
> there any documentation available as to what it does and how to use it?
> 
> Thanks,
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/timn/ros-mongodb_log/pull/2#issuecomment-43640233
> .

rmbennett commented:
Okay awesome - that's great. 

What kind of queries does it support? 

@marc-hanheide

marc-hanheide commented:
Not sure I understand. Any mongo query certainly. But there is a an
interface to convert stuff back to ros. But I'm not the expert. @hawesie
did most the work.

Marc
On 20 May 2014 16:44, "rmbennett" notifications@github.com wrote:

> Okay awesome - that's great.
> 
> What kind of queries does it support?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/timn/ros-mongodb_log/pull/2#issuecomment-43643625
> .

hawesie commented:
Plus @cburbridge too.

The overall README is: https://github.com/strands-project/ros_datacentre/blob/hydro-devel/ros_datacentre/README.md

The main change to mongodb_log is that we've altered some of the format in which the logged messages are stored to allow recreation back into ROS message types and to support a separate 'meta' dictionary from the message content. We've also stripped out the graphing and evaluation code to make it more streamlined. 

We've also added extra mongo-related functionality including the `message_store_node.py` which allows components to write ROS messages directly to the db via a service call (e.g. for event logging in code). See https://github.com/strands-project/ros_datacentre/blob/hydro-devel/ros_datacentre/scripts/example_message_store_client.py for an example.

We will be building on this and maintaining this set of packages for the next 3 years at least (as part of http://strands.acin.tuwien.ac.at) and we'd love to get our code out to the wider community, so it would be great if you could give our code a try for your needs. If you have question you can ask them in the issues: https://github.com/strands-project/ros_datacentre/issues

rmbennett commented:
Thanks very much for the info @hawesie 

I've moved my queries over to the ros_datacentre issue space.

marc-hanheide commented:
this is outdated and superseded by our https://github.com/strands-project/mongodb_store repository. Closing
